index.test.js

```diff
@@ -0,0 +1,283 @@
+ */
+
+const { AssertionFailedError, ValidationFailedError } = require('@uphold/http-errors');
+const createValidator = require('.');
+const asserts = require('validator.js-asserts');
+const _ = require('lodash');
+
+/**
+ * Extra asserts
+ */
+
+function AlwaysValid() {
+  // eslint-disable-next-line no-underscore-dangle
+  this.__class__ = 'AlwaysValid';
+
+  this.validate = function() {
+    return true;
+  };
+
+  return this;
+}
+
+function AlwaysInvalid() {
+  // eslint-disable-next-line no-underscore-dangle
+  this.__class__ = 'AlwaysInvalid';
+
+  this.validate = function() {
+    return false;
+  };
+
+  return this;
+}
+
+/**
+ * Test `Validator`.
+ */
+
+describe('Validator', () => {
+  it('should expose `assert`, `is` and `validate` functions', () => {
+    const { assert, is, validate } = createValidator({
+      AssertionError: AssertionFailedError,
+      ValidationError: ValidationFailedError
+    });
+
+    expect(assert).toBeInstanceOf(Function);
+    expect(is).toBeInstanceOf(Function);
+    expect(validate).toBeInstanceOf(Function);
+  });
+
+  it('should expose custom asserts in pascal case', () => {
+    const { is } = createValidator({
+      AssertionError: AssertionFailedError,
+      ValidationError: ValidationFailedError
+    });
+
+    for (const assert in asserts) {
+      expect(is.prototype).toHaveProperty(assert);
+    }
+  });
+
+  it('should expose extra custom asserts in pascal case', () => {
+    const { is } = createValidator({
+      AssertionError: AssertionFailedError,
+      ValidationError: ValidationFailedError,
+      extraAsserts: { AlwaysInvalid, AlwaysValid }
+    });
+
+    for (const assert in asserts) {
+      expect(is.prototype).toHaveProperty(assert);
+    }
+
+    expect(is.prototype).toHaveProperty('AlwaysValid');
+    expect(is.prototype).toHaveProperty('AlwaysInvalid');
+  });
+
+  describe('assert()', () => {
+    it('should throw an error if assertion fails', () => {
+      const { assert, is } = createValidator({
+        AssertionError: AssertionFailedError,
+        ValidationError: ValidationFailedError
+      });
+
+      try {
+        assert({ name: 'Foo' }, { name: is.integer() });
+
+        fail();
+      } catch (e) {
+        expect(e).toBeInstanceOf(AssertionFailedError);
+        expect(e.errors.name).toHaveLength(1);
+        expect(e.errors.name[0].show().assert).toEqual('Integer');
+      }
+    });
+
+    it('should not mask data by default', () => {
+      const { is, assert } = createValidator({
+        AssertionError: AssertionFailedError,
+        ValidationError: ValidationFailedError
+      });
+
+      const data = assert(
+        {
+          foo: {
+            bar: 'qux',
+            biz: 'bar'
+          },
+          qux: 'biz'
+        },
+        {
+          foo: {
+            bar: is.required(),
+            biz: is.required()
+          }
+        }
+      );
+
+      expect(data).toEqual({ foo: { bar: 'qux', biz: 'bar' }, qux: 'biz' });
```

Comment:
```suggestion
      const data = {
        foo: {
          bar: 'qux',
          biz: 'bar'
        },
        qux: 'biz'
      };
      const returnedData = assert(data, {
        foo: {
          bar: is.required(),
          biz: is.required()
        }
      });

      expect(returnedData).toEqual(data);
```

What do you think about storing the var so it can be clearer to compare in the assertion?

Reply:
Agreed. ✅